Architecture Diagrams (#201)
mulesoft/docs-gateway: gateway/1.4/modules/ROOT/pages/flex-architecture-uc2.adoc

> = Use Case 2: Organization-Owned API Exposed to a Non-organization Owned API Consumer

Reviewer: ```suggestion
= Use Case 2: Organization-Owned API Exposed to a Non-Organization-Owned API Consumer
```

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images

> Use case two covers all patterns of an organization-owned API exposed to a non-organization-owned API consumer, for example, a user making a request to your public facing API.

Reviewer: ```suggestion
Use case two covers all patterns of an organization-owned API exposed to a non-organization-owned API consumer, for example, when a user makes a request to your public-facing API.
```

> Use case two covers all patterns of an organization-owned API exposed to a non-organization-owned API consumer, for example, when a user makes a request to your public-facing API.

Reviewer: ```suggestion
Use case two covers all patterns of a scenario in which an organization-owned API is exposed to a non-organization-owned API consumer; for example, when a user makes a request to your public-facing API.
```

== Pattern 5: Organization Internal API exposed to Non-Organization External Consumer

> In pattern five, the organization owned API is in infrastructure isolated from the API consumer. The consumer can only access the exposed API via the internet. In this use case, you deploy Flex Gateway as an ingress controller.

Reviewer: ```suggestion
In pattern five, the infrastructure of the organization-owned API is isolated from the API consumer. The consumer can only access the exposed API via the internet. In this use case, you deploy Flex Gateway as an ingress controller.
```

> In pattern five, the infrastructure of the organization-owned API is isolated from the API consumer. The consumer can only access the exposed API via the internet. In this use case, you deploy Flex Gateway as an ingress controller.

Reviewer: ```suggestion
In pattern five, the infrastructure of the organization-owned API is isolated from the API consumer. The consumer can access the exposed API only via the internet. In this use case, you deploy Flex Gateway as an ingress controller.
```

image:ad-uc-1-p-2.png[Consumer application requests pass through the internet and a firewall before reaching Flex Gateway.]

> The following diagram shows the physical implementation of an organization internal API exposed to a non-organization external consumer on Kubernetes. The diagram assumes that the firewall terminates the incoming external mTLS connection.

Reviewer: ```suggestion
The following diagram shows the physical implementation of a scenario in which an organization-owned internal API is exposed to a non-organization-owned external consumer on Kubernetes. The diagram assumes that the firewall terminates the incoming external mTLS connection.
```

image:ad-uc-2-p-5-physical.png[A detailed view of pattern five, which contains the necessary services to support Flex gateway]

== Pattern 6: Organization External API exposed to a Non-Organization External Consumer

> In pattern six, Flex Gateway acts as an intermediary between two external network entities. Flex Gateway runs as an ingress and egress controller applying secure communication and policy enforcement.

Reviewer: ```suggestion
In pattern six, Flex Gateway acts as an intermediary between two external network entities. Flex Gateway runs as an ingress and egress controller that applies secure communication and policy enforcement.
```

image:ad-uc-1-p-4.png[Flex Gateway acts as an intermediary between an external application and external API.]

> The following diagram shows the physical implementation of an organization external API exposed to an non-organization external consumer on Kubernetes. The diagram assumes that the firewall terminates the external mTLS connections.

Reviewer: ```suggestion
The following diagram shows the physical implementation of a scenario in which an organization external API is exposed to an non-organization external consumer on Kubernetes. The diagram assumes that the firewall terminates the external mTLS connections.
```
